feat: add yarn modern pnp support (#599)
cypress-io/github-action: README.md

- Use [Yarn Classic](#yarn-classic)
- Use [Yarn Modern](#yarn-modern)
- Use [Yarn workspaces](#yarn-workspaces)

> - Use [Yarn Plug'n'Play](#yarn-plugnplay)

Reviewer: Let's move this section above `Yarn workspaces`.

Author: 🤦‍♂️ Yep, totally should have caught that!

[![Yarn workspaces example](https://github.com/cypress-io/github-action/workflows/example-start-and-yarn-workspaces/badge.svg?branch=master)](.github/workflows/example-start-and-yarn-workspaces.yml)

### Yarn Plug'n'Play

> If you are using a modern yarn (Yarn 2 and later) with [Plug'n'Play enabled](https://yarnpkg.com/features/pnp), you will need to use the `command` parameter to run cypress using yarn dlx instead of npx so the imports are correctly resolved and the action can find the correct cypress command.

Reviewer: ```suggestion
When using [Yarn Modern](https://yarnpkg.com/) (Yarn 2 and later) with [Plug'n'Play](https://yarnpkg.com/features/pnp) enabled, you will need to use the `command` parameter to run [yarn dlx](https://yarnpkg.com/cli/dlx) instead of [npx](https://docs.npmjs.com/cli/v9/commands/npx).
```
